kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryInKjarIntegrationTest.java

```diff
@@ -0,0 +1,235 @@
+
+    private static ClassLoader classLoader = QueryInKjarIntegrationTest.class.getClassLoader();
+
+    private static ReleaseId releaseIdKjarOne = new ReleaseId("org.kie.server.testing", "query-definition-kjar-one", "1.0.0.Final");
+    private static ReleaseId releaseIdKjarOneUpdated = new ReleaseId("org.kie.server.testing", "query-definition-kjar-one", "1.0.1.Final");
+    private static ReleaseId releaseIdKjarTwo = new ReleaseId("org.kie.server.testing", "query-definition-kjar-two", "1.0.0.Final");
+    private static ReleaseId releaseIdKjarDuplicateQueries = new ReleaseId("org.kie.server.testing", "query-definition-kjar-duplicate-queries", "1.0.0.Final");
+    private static ReleaseId releaseIdKjarEmptyQueries = new ReleaseId("org.kie.server.testing", "query-definition-kjar-empty-queries", "1.0.0.Final");
+    private static ReleaseId releaseIdKjarUnmarshallableQueries = new ReleaseId("org.kie.server.testing", "query-definition-kjar-unmarshallable-queries", "1.0.0.Final");
+
+    private static final String KJAR_ONE_REGISTERED_QUERY = "first-query";
+    private static final String KJAR_TWO_REGISTERED_QUERY = "second-query";
+    private static final String KJAR_DUPLICATE_QUERIES_REGISTERED_QUERY = KJAR_TWO_REGISTERED_QUERY;
+    private static final String KJAR_ONE_UPDATED_FIRST_REGISTERED_QUERY = "first-query";
+    private static final String KJAR_ONE_UPDATED_SECOND_REGISTERED_QUERY = KJAR_TWO_REGISTERED_QUERY;
+
+    private static final String EXPECTED_RESOLVED_DS = System.getProperty("org.kie.server.persistence.ds", "jdbc/jbpm-ds");
+
+    @BeforeClass
+    public static void buildAndDeployArtifacts() {
+        KieServerDeployer.createAndDeployKJar(releaseIdKjarOne, getQueryDefinitionKjarContentFromFile("query-definitions-kjar-one.json"));
+        KieServerDeployer.createAndDeployKJar(releaseIdKjarOneUpdated, getQueryDefinitionKjarContentFromFile("query-definitions-kjar-one-updated.json"));
+        KieServerDeployer.createAndDeployKJar(releaseIdKjarTwo, getQueryDefinitionKjarContentFromFile("query-definitions-kjar-two.json"));
+        KieServerDeployer.createAndDeployKJar(releaseIdKjarDuplicateQueries, getQueryDefinitionKjarContentFromFile("query-definitions-kjar-duplicate-queries.json"));
+
+        KieServerDeployer.createAndDeployKJar(releaseIdKjarEmptyQueries, getQueryDefinitionKjarContentFromValue(""));
+        KieServerDeployer.createAndDeployKJar(releaseIdKjarUnmarshallableQueries, getQueryDefinitionKjarContentFromValue("unmarshallable content"));
+    }
+
+    @After
+    public void deleteContainers() {
+        KieContainerResourceList containers = client.listContainers().getResult();
+        for (KieContainerResource container : containers.getContainers()) {
+            client.disposeContainer(container.getContainerId());
+        }
+    }
+
+    @Test
+    public void testQueryDefinitionsFromKjar() throws Exception {
+        String containerOne = "container-one";
+        String containerTwo = "container-two";
```

Comment:
Container ids here have limited relevance, they have a meaning just inside their own test method. So I would rather keep them local, in the tests. 
Updated their naming to better describe their purpose.